In [5]:
df = pd.read_csv("mercedesbenz.csv",usecols=['X0','X1','X3','X4','X5'])

<IPython.core.display.Javascript object>

In [6]:
df.head()


X0 X1 X3 X4 X5
0   k  v  a  d  u
1   k  t  e  d  y
2  az  w  c  d  x
3  az  t  f  d  x
4  az  v  f  d  h

In [13]:
for col in df.columns:
    print(col, ':' ,len(df[col].unique()),'labels')
    

X0 : 47 labels
X1 : 27 labels
X3 : 7 labels
X4 : 4 labels
X5 : 29 labels


In [16]:
pd.get_dummies(df,drop_first=True).shape

<IPython.core.display.Javascript object>

(4209, 109)

to perform one hot encodning we need 109 columns 

In [22]:
df.X1.value_counts().sort_values(ascending=False).head(20)

aa    833
s     598
b     592
l     590
v     408
r     251
i     203
a     143
c     121
o      82
w      52
z      46
u      37
e      33
m      32
t      31
h      29
f      23
y      23
j      22
Name: X1, dtype: int64

In [55]:
top_10 = [x for x in date.X1.value_counts().sort_values(ascending=False).head(10).index]
type(top_10)

list

In [37]:
for label in top_10:
    df[label]=np.where(df['X1']==label,1,0)

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

In [38]:
df[['X1']+top_10].head(20)

X1  aa  s  b  l  v  r  i  a  c  o
0    v   0  0  0  0  1  0  0  0  0  0
1    t   0  0  0  0  0  0  0  0  0  0
2    w   0  0  0  0  0  0  0  0  0  0
3    t   0  0  0  0  0  0  0  0  0  0
4    v   0  0  0  0  1  0  0  0  0  0
5    b   0  0  1  0  0  0  0  0  0  0
6    r   0  0  0  0  0  1  0  0  0  0
7    l   0  0  0  1  0  0  0  0  0  0
8    s   0  1  0  0  0  0  0  0  0  0
9    b   0  0  1  0  0  0  0  0  0  0
10   r   0  0  0  0  0  1  0  0  0  0
11   r   0  0  0  0  0  1  0  0  0  0
12   b   0  0  1  0  0  0  0  0  0  0
13   r   0  0  0  0  0  1  0  0  0  0
14   s   0  1  0  0  0  0  0  0  0  0
15   l   0  0  0  1  0  0  0  0  0  0
16   r   0  0  0  0  0  1  0  0  0  0
17  aa   1  0  0  0  0  0  0  0  0  0
18   c   0  0  0  0  0  0  0  0  1  0
19   a   0  0  0  0  0  0  0  1  0  0

In [48]:
def one_hot(df,variable):
    top_10=[x for x  in df.variable.value_counts().sort_values(ascending=False).head(10).index]
    for label in top_10:
        df[variable+'_'+label]=np.where(data[variable]==label,1,0)
  

In [49]:
one_hot(df,'X3')

AttributeError: 'DataFrame' object has no attribute 'variable'

In [50]:
def one_hot(data, variable,top_x_labels):
    for label in top_x_labels:
        df[variable+'_'+label]=np.where(date[variable]==label,1,0)

In [54]:
df=pd.read_csv('mercedesbenz.csv',usecols=['X1','X2','X3','X4','X5'])
one_hot(df,'X1',top_10)
df

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

X1  X2 X3 X4  X5  X1_aa  X1_s  X1_b  X1_l  X1_v  X1_r  X1_i  X1_a  X1_c  \
0     v  at  a  d   u      0     0     0     0     1     0     0     0     0   
1     t  av  e  d   y      0     0     0     0     0     0     0     0     0   
2     w   n  c  d   x      0     0     0     0     0     0     0     0     0   
3     t   n  f  d   x      0     0     0     0     0     0     0     0     0   
4     v   n  f  d   h      0     0     0     0     1     0     0     0     0   
...  ..  .. .. ..  ..    ...   ...   ...   ...   ...   ...   ...   ...   ...   
4204  s  as  c  d  aa      0     1     0     0     0     0     0     0     0   
4205  o   t  d  d  aa      0     0     0     0     0     0     0     0     0   
4206  v   r  a  d  aa      0     0     0     0     1     0     0     0     0   
4207  r   e  f  d  aa      0     0     0     0     0     1     0     0     0   
4208  r  ae  c  d  aa      0     0     0     0     0     1     0     0     0   

      X1_o  
0        0  
1        0  
2        0  
3        0  
4        0  
...    ...  
4204     0  
4205     1  
4206     0  
4207     0  
4208     0  

[4209 rows x 15 columns]

In [2]:
df = pd.read_csv('mercedesbenz.csv',usecols=['X1','X2','X3','X4','X5'])

<IPython.core.display.Javascript object>

In [3]:
for col in df.columns:
    print(col,' : ',len(df[col].unique()),'labels')

X1  :  27 labels
X2  :  44 labels
X3  :  7 labels
X4  :  4 labels
X5  :  29 labels


In [5]:
pd.get_dummies(df,drop_first=True).shape

<IPython.core.display.Javascript object>

(4209, 106)

In [12]:
top_10 = [x for x in df.X2.value_counts().sort_values(ascending=False).head(10).index]
top_10

['as', 'ae', 'ai', 'm', 'ak', 'r', 'n', 's', 'f', 'e']

In [16]:
for label in top_10:
    df[label] =np.where(df['X2']==label,1,0)

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

In [17]:
df['X2'+top_10]

TypeError: can only concatenate str (not "list") to str

In [3]:
data = pd.read_csv('mercedesbenz.csv',usecols=['X0','X1','X2','X3','X4','X5'])

<IPython.core.display.Javascript object>

In [8]:
for col in data.columns:
    print(col,' :' , len(data[col].unique()) , 'labels')

X0  : 47 labels
X1  : 27 labels
X2  : 44 labels
X3  : 7 labels
X4  : 4 labels
X5  : 29 labels


In [11]:
data.shape

(4209, 6)

In [10]:
pd.get_dummies(data,drop_first= True).shape

<IPython.core.display.Javascript object>

(4209, 152)

In [20]:
data.X2.value_counts().sort_values(ascending=False).head(20)

as    1659
ae     496
ai     415
m      367
ak     265
r      153
n      137
s       94
f       87
e       81
aq      63
ay      54
a       47
t       29
k       25
i       25
b       21
ao      20
ag      19
z       19
Name: X2, dtype: int64

In [27]:
top_10=[ x for x in data.X2.value_counts().sort_values(ascending=False).head(10).index]
top_10

['as', 'ae', 'ai', 'm', 'ak', 'r', 'n', 's', 'f', 'e']

In [31]:
for lable in top_10:
    data[lable]= np.where(data['X2']==lable,1,0)

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

In [35]:
data[['X2']+top_10]

X2  as  ae  ai  m  ak  r  n  s  f  e
0     at   0   0   0  0   0  0  0  0  0  0
1     av   0   0   0  0   0  0  0  0  0  0
2      n   0   0   0  0   0  0  1  0  0  0
3      n   0   0   0  0   0  0  1  0  0  0
4      n   0   0   0  0   0  0  1  0  0  0
...   ..  ..  ..  .. ..  .. .. .. .. .. ..
4204  as   1   0   0  0   0  0  0  0  0  0
4205   t   0   0   0  0   0  0  0  0  0  0
4206   r   0   0   0  0   0  1  0  0  0  0
4207   e   0   0   0  0   0  0  0  0  0  1
4208  ae   0   1   0  0   0  0  0  0  0  0

[4209 rows x 11 columns]

In [36]:
def one_hot (df,variable , top_x_lable):
    for lable in top_x_lable:
        df[variable+'_'+lable] = np.where(data[variable]==lable,1,0)

In [47]:
top_10inX3=[x for x in data.X3.value_counts().sort_values(ascending=False).head().index]

In [53]:
one_hot(data,'X3',top_10inX3)

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

In [51]:
data=pd.read_csv('mercedesbenz.csv',usecols=['X3'])

<IPython.core.display.Javascript object>

In [54]:
data


X3  X3_c  X3_f  X3_a  X3_d  X3_g
0     a     0     0     1     0     0
1     e     0     0     0     0     0
2     c     1     0     0     0     0
3     f     0     1     0     0     0
4     f     0     1     0     0     0
...  ..   ...   ...   ...   ...   ...
4204  c     1     0     0     0     0
4205  d     0     0     0     1     0
4206  a     0     0     1     0     0
4207  f     0     1     0     0     0
4208  c     1     0     0     0     0

[4209 rows x 6 columns]

In [58]:
df = pd.read_csv('mercedesbenz.csv',usecols=['X1','X2'])
df.shape


<IPython.core.display.Javascript object>

(4209, 2)

## One hot encoding

In [62]:
pd.get_dummies(df,drop_last=True).shape

<IPython.core.display.Javascript object>

TypeError: get_dummies() got an unexpected keyword argument 'drop_last'